app/knight.py

```diff
@@ -0,0 +1,30 @@
+from __future__ import annotations
+
+
+class Knight:
+    def __init__(self, knight: dict) -> None:

```

Comment:
The `__init__` method seems to be doing multiple things, which violates the Single Responsibility Principle. It is better to split this method into smaller ones, each with its own responsibility. For instance, you could create separate methods for calculating the power, hp, and protection of the knight.